src/domain/noteSettings.service.ts

```diff
@@ -99,4 +107,32 @@ export default class NoteService {
   public async patchMemberRoleByUserId(id: NoteId, userId: UserId, newRole: MemberRole): Promise<MemberRole> {
     return await this.noteSettingsRepository.patchMemberRoleByUserId(id, userId, newRole);
   }
+
+  /**
+   * Set new parent for the note
+   *
+   * @param id - Note id
+   * @param parentURL - link to the new parent note
+   */
+  public async updateParent(id: NoteId, parentURL: string): Promise<void> {
+    // The regex matches a substring in the URL that follows '/note/' and consists of exactly 10 characters
+    const regex = /\/note\/([a-zA-Z0-9-_]{10})/;
+
+    const matches = parentURL.match(regex);
+
+    if (matches === null) {
+      throw new Error('Invalid parent URL');
+    }
```

Comment:
can we check all characters after /note/ and throw error if there are more than 10